index.bs

```diff
@@ -5837,108 +5845,164 @@ The [=type name=] of the
 {{boolean}} type is "<code>Boolean</code>".
 
 
-<h4 oldids="dom-byte" id="idl-byte" interface>byte</h4>
+<h4 id="idl-int8" oldids="dom-byte,idl-byte" interface>int8</h4>
 
-The {{byte}} type is a signed integer
+The {{int8}} type is a signed integer
 type that has values in the range [−128, 127].
 
-{{byte}} constant values in IDL are
+{{int8}} constant values in IDL are
 represented with <emu-t class="regex"><a href="#prod-integer">integer</a></emu-t>
 tokens.
 
 The [=type name=] of the
-{{byte}} type is "<code>Byte</code>".
+{{int8}} type is "<code>Int8</code>".
 
+For legacy compatibility, <code>byte</code> is to be treated
+as an alias for {{int8}}, similar to:
 
-<h4 oldids="dom-octet" id="idl-octet" interface>octet</h4>
+<pre>
+    typedef (int8) byte;
+</pre>
+
+
+<h4 id="idl-uint8" oldids="dom-octet,idl-octet" interface>uint8</h4>
 
-The {{octet}} type is an unsigned integer
+The {{uint8}} type is an unsigned integer
 type that has values in the range [0, 255].
 
-{{octet}} constant values in IDL are
+{{uint8}} constant values in IDL are
 represented with <emu-t class="regex"><a href="#prod-integer">integer</a></emu-t>
 tokens.
 
 The [=type name=] of the
-{{octet}} type is "<code>Octet</code>".
+{{uint8}} type is "<code>Uint8</code>".
+
+For legacy compatibility, <code>octet</code> is to be treated
+as an alias for {{uint8}}, similar to:
+
+<pre>
+    typedef (uint8) octet;
+</pre>
 
 
-<h4 oldids="dom-short" id="idl-short" interface>short</h4>
+<h4 id="idl-int16" oldids="dom-short,idl-short" interface>int16</h4>
 
-The {{short}} type is a signed integer
+The {{int16}} type is a signed integer
 type that has values in the range [−32768, 32767].
 
-{{short}} constant values in IDL are
+{{int16}} constant values in IDL are
 represented with <emu-t class="regex"><a href="#prod-integer">integer</a></emu-t>
 tokens.
 
 The [=type name=] of the
-{{short}} type is "<code>Short</code>".
+{{int16}} type is "<code>Int16</code>".
 
+For legacy compatibility, <code>short</code> is to be treated
+as an alias for {{int16}}, similar to:
 
-<h4 oldids="dom-unsignedshort" id="idl-unsigned-short" interface>unsigned short</h4>
+<pre>
+    typedef (int16) short;
+</pre>
+
+
+<h4 id="idl-uint16" oldids="dom-unsignedshort,idl-unsigned-short" interface>uint16</h4>
 
-The {{unsigned short}} type is an unsigned integer
+The {{uint16}} type is an unsigned integer
 type that has values in the range [0, 65535].
 
-{{unsigned short}} constant values in IDL are
+{{uint16}} constant values in IDL are
 represented with <emu-t class="regex"><a href="#prod-integer">integer</a></emu-t>
 tokens.
 
 The [=type name=] of the
-{{unsigned short}} type is "<code>UnsignedShort</code>".
+{{uint16}} type is "<code>Uint16</code>".
+
+For legacy compatibility, <code>unsigned short</code> is to be treated
+as an alias for {{uint16}}, similar to:
+
+<pre>
+    typedef (uint16) unsigned short;
```

Comment:
I believe this causes syntax error, and not sure it's really needed. Each parser implementation can support both form while autofixes by @autokagami land.

Reply:
The “similar to” phrasing may have intended to acknowledge that — but it still seems confusing to have pseudo-idl in the idl spec itself (not very friendly for folks not already familiar enough to know it’s not meant to be valid).

Given the original productions are still explicitly part of the grammar in this PR, I don’t think this framing is needed anyway.